test/infrastructure/inmemory/internal/server/api/handler.go

```diff
@@ -313,16 +314,24 @@ func (h *apiServerHandler) apiV1List(req *restful.Request, resp *restful.Respons
 		listOpts = append(listOpts, client.InNamespace(req.PathParameter("namespace")))
 	}
 
-	// TODO: The only field selector which works is for `spec.nodeName` on pods.
-	selector, err := fields.ParseSelector(req.QueryParameter("fieldSelector"))
+	// TODO: The only field Selector which works is for `spec.nodeName` on pods.
+	fieldSelector, err := fields.ParseSelector(req.QueryParameter("fieldSelector"))
 	if err != nil {
 		_ = resp.WriteErrorString(http.StatusInternalServerError, err.Error())
 		return
 	}
-	if selector != nil {
-		listOpts = append(listOpts, client.MatchingFieldsSelector{Selector: selector})
+	if fieldSelector != nil {
```

Comment:
This is incidental, but I fixed it up in the course of the PR and decided to keep it. Can split it out if needed.